openvpn/fetch-external-configs.sh

```diff
@@ -0,0 +1,42 @@
+#! /bin/bash
+
+VPN_CONFIG_SOURCE_TYPE="${VPN_CONFIG_SOURCE_TYPE:-github_zip}"
+
+# Set default GitHub config repo
+GITHUB_CONFIG_SOURCE_REPO="${GITHUB_CONFIG_SOURCE_REPO:-haugene/vpn-configs}"
+GITHUB_CONFIG_SOURCE_REVISION="${GITHUB_CONFIG_SOURCE_REVISION:-main}"
+
+if [[ "${VPN_CONFIG_SOURCE_TYPE}" == "github_zip" ]]; then
+
+  # Concatenate URL for config bundle from the given GitHub repo
+  GITHUB_CONFIG_BUNDLE_URL="https://github.com/${GITHUB_CONFIG_SOURCE_REPO}/archive/${GITHUB_CONFIG_SOURCE_REVISION}.zip"
+  
+  # Create a temporary file and download bundle to it
+  config_repo_temp_zip_file=$(mktemp)
+  echo "Downloading configs from ${GITHUB_CONFIG_BUNDLE_URL} into ${config_repo_temp_zip_file}"
+  curl -sSL -o "${config_repo_temp_zip_file}" "${GITHUB_CONFIG_BUNDLE_URL}"
```

Comment:
missing check of curl success

Reply:
what are you thinking here? do a `if [ $? -ne 0 ]; then` something? Or postfix the line with `|| exit 1` maybe? Or we could do a `set -e` to fail on any error? But we source the script from start.sh so setting -e might then leak into that script?

Reply:
As much as I dislike `set -e`, I've grown used to our team's standard shell script top of:

```
set -o nounset
set -o errexit
set -o pipefail
```

It might be a matter of taste, but that's why I don't like sourcing one script from another in this type of code, it's a bit like Basic `GOSUB` if you remember that. (https://hwiegman.home.xs4all.nl/gw-man/GOSUB.html) - it makes every variable a global variable.

Does this script HAVE to be sourced (with bash `source`/`.`) in order to set any variables or change other bash internal state (e.g. `cd ...`), or can it be executed as a sub-process?

Reply:
Good question. I'm used to sourcing them in this project because the sub scripts usually set environment variables. But I don't think we would need that here. So then add that header and do the cleanup in a trap - the end?

I've gotten some bash experience over the years and I'm capable. But I'm not very well versed in best practices. So I'm happy to get some pointers :+1:

Reply:
I prefer passing parameters to sub scripts instead of having them rely on environment variables (even if it means passing the environment variable as a parameter). This makes the sub script call more explicit and readable as well as makes running the sub script independently easier (e.g. while developing, or if you find that you want to use it from another point in the code).

Reply:
I'm planning a bigger rewrite at some point. Been coding a bit more Python lately and I think that a lot of the code here could be done cleaner and easier with Python. To take this one step at a time I don't think I'll get away from the env vars any time soon. The project is too reliant on them already. So I'll let it be for this script as well.

But I've added
```
set -o nounset
set -o errexit
set -o pipefail
```
as suggested. and also added `errexit` and `pipefail` to the main script. I'm not sure if it can stay there. We'll have to see how much issues it creates. There are steps that I've seen failing in many logs where the container manages without. Like when people mount /dev/net/tun and don't set CREATE_TUN_DEVICE=false. Then we get a "file exists" type of error but it's fine.

It is probably better to fail fast and add ignores for those cases where a failure is accepted or tolerated in certain scenarios. I'm marking this as resolved for now.

If you have time for another review as it stands now I'd be grateful. But I think it's closing in on merging to `dev` at least. And planning to do some hardening there. And updating documentation etc.